src/lib.rs

```diff
@@ -174,13 +192,24 @@ impl AstarteSdk {
         Ok(())
     }
 
+    pub async fn update_interfaces(
+        &self,
+        interfaces: interfaces::Interfaces,
+    ) -> Result<(), AstarteError> {
+        self.opts.write().await.interfaces = interfaces;
+
+        self.subscribe().await?;
+        self.send_introspection().await?;
```

Comment:
When introspection update at runtime is allowed, some annoying corner cases appear. The solution used is a number of SDKs is to persist the current introspection and save the new one only when it's acknowledged correctly by Astarte.
You can take a look at the [Qt5 SDK](https://github.com/astarte-platform/astarte-device-sdk-qt5/commit/a521af3e1160bc6ad6aeb43a99b4e99473bb042f) or the [Go SDK](https://github.com/astarte-platform/astarte-device-sdk-go/pull/33) for this.

Reply:
this is blocked by this https://github.com/bytebeamio/rumqtt/issues/349 temporarily